### PR TITLE
systemctl: allow disable on the unit file path, but warn about it

### PR DESCRIPTION
**Pull request by msekletar**
systemd now returns an error when it is asked to perform disable on the
unit file path. In the past this was allowed, but systemd never really
considered an actual content of the [Install] section of the unit
file. Instead it performed disable on the unit name, i.e. purged all
symlinks pointing to the given unit file (undo of implicit link action
done by systemd when enable is called on the unit file path) and all
symlinks that have the same basename as the given unit file.

However, to notice that [Install] info of the file is not consulted one
must create additional symlinks manually. I argue that in most cases
users do not create such links. Let's be nice to our users and don't
break existing scripts that expect disable to work with the unit file
path.

### CODE REVIEW

**Comment by poettering** [comment]
lgtm

**Comment by poettering** [comment]
not sure what happened to semaphore, but i figure if the ubuntu CIs like it that good enoguh, too.

File: src/systemctl/systemctl.c [hunk 1]
```diff
@@ -5670,6 +5670,32 @@ static int mangle_names(char **original_names, char ***mangled_names) {
...
+
+                if (!is_path(*u))
+                        continue;
+
+                n = strdup(basename(*u));
+                if (!n)
+                        return log_oom();
+
+                *u = mfree(*u);
+                *u = n;
```
Comment (keszybz) [inline src/systemctl/systemctl.c]:
```
r = free_and_strdup(u, basename(*u));
if (r < 0)
       return log_error_errno(...);
```

**Comment by keszybz** [comment]
LvGTMt, but needs a style fix. That `mfree` is just too ugly :)

**Comment by msekletar** [comment]
@keszybz force pushed now.

**Pull request merged**